libs/sr/db/src/Database/DataManagers/TeamDataManager.ts

```diff
@@ -1,33 +1,59 @@
 import {
-  deepClone,
   objKeyMap,
   objMap,
+  pruneOrPadArray,
   range,
+  shallowCompareObj,
 } from '@genshin-optimizer/common/util'
-import type { RelicSlotKey } from '@genshin-optimizer/sr/consts'
-import { allRelicSlotKeys } from '@genshin-optimizer/sr/consts'
+import type { CharacterKey, RelicSlotKey } from '@genshin-optimizer/sr/consts'
+import {
+  allCharacterKeys,
+  allRelicSlotKeys,
+} from '@genshin-optimizer/sr/consts'
+import type { Member, Sheet, Tag } from '@genshin-optimizer/sr/formula'
 import type { ICachedRelic } from '../../Interfaces'
 import { DataManager } from '../DataManager'
 import type { SroDatabase } from '../Database'
+import { validateTag } from '../tagUtil'
 
 const buildTypeKeys = ['equipped', 'real', 'tc'] as const
 type BuildTypeKey = (typeof buildTypeKeys)[number]
-export type LoadoutMetadatum = {
-  loadoutId: string
+export type TeamMetadatum = {
+  characterKey: CharacterKey
+
   buildType: BuildTypeKey
   buildId: string
   buildTcId: string
   compare: boolean
   compareType: BuildTypeKey
   compareBuildId: string
   compareBuildTcId: string
+
+  optConfigId?: string
 }
 export interface Team {
   name: string
   description: string
 
-  loadoutMetadata: Array<LoadoutMetadatum | undefined>
   lastEdit: number
+
+  // frames, store data as a "sparse 2d array"
+  frames: Array<Tag>
+  conditionals: Array<{
+    sheet: Sheet
```

Comment:
should this be stored as a Tag, same as bonusstats?

Reply:
its pretty much a tag + value, this just makes the values more explicit since the fields are optional in a tag, but necessary in a conditional